feat(HTTP): Refresh 1XX response pages (#35023)
mdn/content: files/en-us/web/http/status/103/index.md

{{HTTPSidebar}}

The HTTP **`103 Early Hints`** [informational response](/en-US/docs/Web/HTTP/Status#information_responses) may be sent by a server while it is still preparing a response, with hints about the sites and resources that the server expects the final response will link to.
This allows a browser to [preconnect](/en-US/docs/Web/HTML/Attributes/rel/preconnect) to sites or start [preloading](/en-US/docs/Web/HTML/Attributes/rel/preload) resources even before the server has prepared and sent a final response.

The early hint response is primarily intended for use with the {{HTTPHeader("Link")}} header, which indicates the resources to be loaded.
It may also contain a [`Content-Security-Policy`](/en-US/docs/Web/HTTP/CSP) header that is enforced while processing the early hint.

A server might send multiple `103` responses, for example, following a redirect.
Browsers only process the first early hint response, and this response must be discarded if the request results in a cross-origin redirect.

> Preloaded resources from the early hint are effectively prepended to the `Document`'s head element, and then followed by the resources loaded in the final response.

Reviewer: or this?

```suggestion
Preloaded resources from the early hint are effectively prepended to the `Document`'s `<head>` element, and then followed by the resources loaded in the final response.
```

Author: I've removed in https://github.com/mdn/content/pull/35023/commits/730c3140fd4ef75d2f1e8c5378a6d836cecd5c51

> Preloaded resources from the early hint are effectively prepended to the `Document`'s head element, and then followed by the resources loaded in the final response.

Reviewer: Does it really mean "head element"? If yes, then add `<>` to head or did it mean headers?:
```suggestion
Preloaded resources from the early hint are effectively prepended to the `Document`'s headers and then followed by the resources loaded in the final response.
```

Author: Well, I think this is not literal, it's a way for readers to conceptualize what's happening, I will reword or maybe remove

Author: I think this was a bit confusing, so it's likely better off without it -> https://github.com/mdn/content/pull/35023/commits/730c3140fd4ef75d2f1e8c5378a6d836cecd5c51

> **Note:**

> > For compatibility reasons [it is recommended](https://www.rfc-editor.org/rfc/rfc8297#section-3) to only send HTTP `103 Early Hints` responses over HTTP/2 or later, unless the client is known to handle informational responses correctly.

Reviewer: ```suggestion
> For compatibility and security reasons, it is recommended to [only send HTTP `103 Early Hints` responses over HTTP/2 or later]((https://www.rfc-editor.org/rfc/rfc8297#section-3) unless the client is known to handle informational responses correctly.
```
For a11y reasons, prefer to link "SEO interesting" terms.

## See also

- {{HTTPHeader("Link")}}
- [Cross-Origin Resource Sharing (CORS)](/en-US/docs/Web/HTTP/CORS)
- [Content Security Policy (CSP)](/en-US/docs/Web/HTTP/CSP)

> - [rel=preconnect](/en-US/docs/Web/HTML/Attributes/rel/preconnect) ({{htmlelement("link")}} element attribute)
> - [rel=preload](/en-US/docs/Web/HTML/Attributes/rel/preload) ({{htmlelement("link")}} element attribute)

Reviewer: ```suggestion
- [`rel="preconnect"`](/en-US/docs/Web/HTML/Attributes/rel/preconnect) ({{htmlelement("link")}} attribute)
- [`rel="preload"`](/en-US/docs/Web/HTML/Attributes/rel/preload) ({{htmlelement("link")}} attribute)
```
the link will be encased in `<>` so i think we can drop the word 'element'
